module_utils/ansible_nailgun_cement.py

```diff
@@ -31,6 +32,107 @@
 from nailgun import entity_mixins, entity_fields
 
 
+class TemplateInput(
+    Entity,
+    entity_mixins.EntityCreateMixin,
+    entity_mixins.EntityDeleteMixin,
+    entity_mixins.EntityReadMixin,
+    entity_mixins.EntitySearchMixin,
+    entity_mixins.EntityUpdateMixin
+):
+
+    def __init__(self, server_config=None, **kwargs):
+        _check_for_value('template', kwargs)
+        self._fields = {
+            'advanced': entity_fields.BooleanField(),
+            'description': entity_fields.StringField(),
+            'fact_name': entity_fields.StringField(),
+            'input_type': entity_fields.StringField(),
+            'name': entity_fields.StringField(),
+            'options': entity_fields.StringField(),
+            'puppet_parameter_class': entity_fields.StringField(),
+            'puppet_parameter_name': entity_fields.StringField(),
+            'required': entity_fields.BooleanField(),
+            # There is no Template base class yet
+            'template': entity_fields.OneToOneField(JobTemplate, required=True),
+            'variable_name': entity_fields.StringField(),
+        }
+        super(TemplateInput, self).__init__(server_config, **kwargs)
+        self._meta = {
+            'api_path': '/api/v2/templates/{0}/template_inputs'.format(self.template.id),
+            'server_modes': ('sat')
+        }
+
+    def read(self, entity=None, attrs=None, ignore=None, params=None):
+        if entity is None:
+            entity = TemplateInput(template=self.template)
+        if ignore is None:
+            ignore = set()
+        ignore.add('advanced')
+        ignore.add('puppet_parameter_class')
+        return super(TemplateInput, self).read(entity=entity, attrs=attrs, ignore=ignore, params=params)
+
+
+class JobTemplate(
+    Entity,
+    entity_mixins.EntityCreateMixin,
+    entity_mixins.EntityDeleteMixin,
+    entity_mixins.EntityReadMixin,
+    entity_mixins.EntitySearchMixin,
+    entity_mixins.EntityUpdateMixin
+):
+    """A representation of a Job invocation entity."""
+
+    def __init__(self, server_config=None, **kwargs):
+        self._fields = {
+            'audit_comment': entity_fields.StringField(),
+            'description_format': entity_fields.StringField(),
+            'effective_user': entity_fields.DictField(),
+            'job_category': entity_fields.StringField(),
+            'location': entity_fields.OneToManyField(Location),
+            'locked': entity_fields.BooleanField(),
+            'name': entity_fields.StringField(),
+            'organization': entity_fields.OneToManyField(Organization),
+            'provider_type': entity_fields.StringField(),
+            'snippet': entity_fields.BooleanField(),
+            'template': entity_fields.StringField(),
+            'template_inputs': entity_fields.OneToManyField(TemplateInput),
+        }
+        self._meta = {
```

Comment:
`_meta` is called before `super()` here like in any other entity in nailgun. In case of TemplateInput `super()` has to be called first, as the `self.template.id` is used in the meta block.